core/src/main/java/com/scalar/db/service/FactoryManager.java

```diff
@@ -0,0 +1,119 @@
+package com.scalar.db.service;
+
+import com.google.common.collect.ImmutableMap;
+import com.scalar.db.api.DistributedStorage;
+import com.scalar.db.api.DistributedStorageAdmin;
+import com.scalar.db.api.DistributedStorageFactory;
+import com.scalar.db.api.DistributedTransactionAdmin;
+import com.scalar.db.api.DistributedTransactionFactory;
+import com.scalar.db.api.DistributedTransactionManager;
+import com.scalar.db.api.TwoPhaseCommitTransactionManager;
+import com.scalar.db.config.DatabaseConfig;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+/**
+ * This class loads {@link DistributedStorageFactory} and {@link DistributedTransactionFactory} with
+ * {@link ServiceLoader}.
+ */
+final class FactoryManager {
```

Comment:
This class loads the implementations of `DistributedStorageFactory` and `DistributedTransactionFactory` with `ServiceLoader`.